dash/dash.py

```diff
@@ -217,6 +225,24 @@ def add_url(name, view_func, methods=('GET',)):
         self._cached_layout = None
         self.routes = []
 
+        # hot reload
+        self._reload_hash = _generate_hash() \
+            if hot_reload else ''
+        self._reload_interval = hot_reload_interval
+        self._hard_reload = False
+        self._lock = threading.RLock()
+        self._watch_thread = None
+        if hot_reload:
+            self._watch_thread = threading.Thread(
+                target=lambda: _watch.watch([self._assets_folder],
+                                            self._on_assets_change,
+                                            sleep_time=0.5))
+            self._watch_thread.daemon = True
+            self._watch_thread.start()
+
+            if not self.server.debug:
+                self.server.debug = True
```

Comment:
I think this would get over-ridden when somebody calls `app.run_server(debug=False)`. Should that happen? I think in `run_server` you can just say `debug = debug or self.hot_reload`

Reply:
I think I will remove that from here and add it in the enable method from #369.